File: src/main/java/vergisst/minecraftmod/weaponthrow/packets/BasePacket.java
package vergisst.minecraftmod.weaponthrow.packets;

import io.netty.buffer.Unpooled;
import net.minecraft.network.PacketByteBuf;
import net.minecraft.util.Identifier;

public abstract class BasePacket {
    private Identifier identifier;

    protected PacketByteBuf buf;

    public BasePacket(Identifier id) {
        identifier = id;
        buf = new PacketByteBuf(Unpooled.buffer());
    }

    public PacketByteBuf getBuf() {
        return buf;
    }

    public Identifier getIdentifier() {
        return identifier;
    }
}

File: src/main/kotlin/vergisst/minecraftmod/weaponthrowlite/packets/BasePacket.kt
package vergisst.minecraftmod.weaponthrowlite.packets

import io.netty.buffer.Unpooled
import net.minecraft.network.PacketByteBuf
import net.minecraft.util.Identifier

abstract class BasePacket(val identifier: Identifier?) {
    var buf: PacketByteBuf?
        protected set

    init {
        this.buf = PacketByteBuf(Unpooled.buffer())
    }
}
